### PR TITLE
Improved with NoSuchEntityException catch for item in stock

### PR DESCRIPTION
**Pull request by winniepukki**

### CODE REVIEW

**Comment by winniepukki** [comment]
Referencing [the original](https://github.com/scandipwa/scandipwa/issues/2096) issue

File: src/Model/Resolver/EntityUrl.php [hunk 1]
```diff
@@ -139,7 +139,14 @@ public function resolve(
...
                 $product = $collection->addIdFilter($id)->getFirstItem();
-                $isInStock = $this->stockItemRepository->get($id)->getIsInStock();
+                $isInStock = false;
+
+                try {
+                    $isInStock = $this->stockItemRepository->get($id)->getIsInStock();
+                    throw new NoSuchEntityException();
+                } catch (NoSuchEntityException $e) {
+                    // Ignoring error is safe
+                }
```
Comment (alfredsgenkins) [inline src/Model/Resolver/EntityUrl.php]: What is going on here? So you throw the error manually to achieve what? This seem to me to work OK even without doing that `throw new NoSuchEntityException();`.

**Review by alfredsgenkins — CHANGES_REQUESTED** [review]

File: src/Model/Resolver/EntityUrl.php [hunk 2]
```diff
@@ -139,7 +139,13 @@ public function resolve(
                 $collection = $this->productCollectionFactory->create()
                     ->addAttributeToFilter('status', ['eq' => Status::STATUS_ENABLED]);
                 $product = $collection->addIdFilter($id)->getFirstItem();
-                $isInStock = $this->stockItemRepository->get($id)->getIsInStock();
+                $isInStock = false;
```
Comment (AleksandrsKondratjevs) [inline src/Model/Resolver/EntityUrl.php:142]: Hey!
If stockItemRepository fails to get stock status by product id is it really means that this product is out of stock?

Comment (winniepukki) [reply src/Model/Resolver/EntityUrl.php:142]: @AleksandrsKondratjevs Technically yes. If we are unable to get the product, how can we verify the stock?

File: src/Model/Resolver/EntityUrl.php [hunk 3]
```diff
@@ -139,7 +139,13 @@ public function resolve(
...
+
+                try {
+                    $isInStock = $this->stockItemRepository->get($id)->getIsInStock();
+                } catch (NoSuchEntityException $e) {
+                    // Ignoring error is safe
```
Comment (alfredsgenkins) [inline src/Model/Resolver/EntityUrl.php:147]: Instead of ignoring the error I would `return null;` but it is a preference. Merging.

**Review by alfredsgenkins — APPROVED** [review]

**Pull request merged**